src/App.tsx

```diff
@@ -1,24 +1,225 @@
+} from 'react';
 import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+import { Todo } from './types/Todo';
+import { getpreparedTodos } from './utils/PreparedTodos';
+import { TodosFilter } from './types/TodoFilter';
+import {
+  addTodos,
+  deleteTodos,
+  getTodos,
+  updateTodo,
+} from './api/todos';
+import { TodoApp } from './components/TodoApp/TodoApp';
+import { TodoHeader } from './components/TodoHeader/TodoHeader';
+import { ErrorMessage } from './components/ErrorMessage/ErrorMessage';
+import { TempTodo } from './components/TempTodo/TempTodo';
+import { TodoFilter } from './components/TodoFilter/TodoFilter';
+import { USER_ID } from './utils/idConstant';
+import { ErrorMessages } from './utils/errorMessage';
 
 export const App: React.FC = () => {
+  const [todos, setTodo] = useState<Todo[]>([]);
+  const [filter, setFilter] = useState<TodosFilter>(TodosFilter.All);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [isLoading, setIsLoading] = useState(false);
+  const [loadingId, setLoadingId] = useState<number[]>([]);
+  const [isLoaderActive, setIsLoaderActive] = useState(false);
+  const [todoTitle, setTitle] = useState('');
+
+  const activeTodosCount = todos
+    .filter(({ completed }) => completed === false).length;
+
+  const hasCompletedTodosCount = todos.filter(todo => todo.completed).length;
+
+  useEffect(() => {
+    setIsLoading(true);
+    getTodos(USER_ID)
+      .then(createdTodo => {
+        setTodo(createdTodo);
+      })
+      .catch(() => {
+        setErrorMessage(ErrorMessages.LoadError);
+      })
+      .finally(() => {
+        setIsLoading(false);
+      });
+  }, []);
+
+  const handleAddTodos = (newTodo: Omit<Todo, 'id'>) => {
+    setLoadingId([0]);
+    addTodos(newTodo)
+      .then((createdTodo) => {
+        setTodo((currentTodo) => [...currentTodo, createdTodo]);
+        setTitle('');
+      })
+      .catch(() => {
+        setErrorMessage(ErrorMessages.AddError);
+      })
+      .finally(() => {
+        setTempTodo(null);
+        setIsLoading(false);
+        setLoadingId([]);
+      });
+    const temp: Todo = Object.assign(newTodo, { id: 0 });
+
+    setTempTodo(temp);
+  };
+
+  const handleDelete = (todoId: number) => {
+    setLoadingId([todoId]);
+    setIsLoaderActive(true);
+    deleteTodos(todoId)
+      .then(() => {
+        setTodo(prevTodos => prevTodos.filter(todo => todo.id !== todoId));
+        setErrorMessage('');
+      })
+      .catch(() => {
+        setErrorMessage(ErrorMessages.DeleteError);
+      })
+      .finally(() => {
+        setLoadingId([]);
+        setIsLoading(false);
+        setIsLoaderActive(false);
+      });
+  };
+
+  const handleClearCompleted = () => {
+    const isSomeTodosCompleted = todos
+      .reduce((total: number[], current) => (current.completed
+        ? [...total, current.id]
+        : total), []);
+
+    Promise.all(isSomeTodosCompleted)
```

Comment:
Read how `Promise.all` works - https://developer.mozilla.org/ru/docs/Web/JavaScript/Reference/Global_Objects/Promise/all
Now it doesn't make sense what are you doing